clang/lib/AST/ByteCode/Compiler.cpp

```diff
@@ -5452,6 +5477,11 @@ bool Compiler<Emitter>::visitSwitchStmt(const SwitchStmt *S) {
   if (!this->emitSetLocal(CondT, CondVar, S))
     return false;
 
+  if (auto *DD =
+          dyn_cast_if_present<DecompositionDecl>(S->getConditionVariable()))
+    if (!this->emitDecompositionVarInit(DD))
+      return false;
```

Comment:
Nit:
```suggestion
  if (auto *DD =
          dyn_cast_if_present<DecompositionDecl>(S->getConditionVariable()); DD && !this->emitDecompositionVarInit(DD))
      return false;
```

Maybe also deduplicate with the above.